diff -urN sane-backends-1.0.7.orig/backend/plustek_driver/Makefile sane-backends-1.0.7/backend/plustek_driver/Makefile
--- sane-backends-1.0.7.orig/backend/plustek_driver/Makefile	Tue Mar 12 22:46:19 2002
+++ sane-backends-1.0.7/backend/plustek_driver/Makefile	Tue Mar 12 22:49:58 2002
@@ -39,8 +39,10 @@
 #
 # Comment out if you are not running SMP. Someone take this out of here
 # when the SMP stuff gets moved out of the kernel Makefile.
-# SMP = 1
-# SMP_PROF = 1
+ifdef BUILD_SMP
+SMP = 1
+SMP_PROF = 1
+endif
 
 #
 # add the following to get assembly listing
@@ -72,9 +72,9 @@
 endif
 
 ifeq ($(DEBUG),y)
-  DEBFLAGS = -O -g -DDEBUG # "-O" is needed to expand inlines
+  DEBFLAGS = $(OPT_FLASG) -DDEBUG # "-O" is needed to expand inlines
 else
-  DEBFLAGS = -O2
+  DEBFLAGS = $(OPT_FLAGS)
 endif
 
 #
@@ -118,7 +118,7 @@
 MODLIB    = /lib/modules/$(LINUXVERSION)
 
 ifdef SMP
-CFLAGS += -D__SMP__
+CFLAGS += -D__SMP__ -DCONFIG_X86_LOCAL_APIC
 
 ifdef SMP_PROF
 CFLAGS += -D__SMP_PROF__
